File: java-reactive-ms/domain/usecase/src/main/java/com/maocq/usecase/gethello/GetHelloUseCase.java
package com.maocq.usecase.gethello;

import com.maocq.model.hello.gateways.HelloRepository;
import lombok.RequiredArgsConstructor;
import reactor.core.publisher.Mono;
import reactor.util.function.Tuple2;

@RequiredArgsConstructor
public class GetHelloUseCase {

    private final HelloRepository helloRepository;

    public Mono<String> hello(int latency) {
        return helloRepository.hello(latency);
    }

    public Mono<String> multipleHello() {
        return helloRepository.hello(50)
          .flatMap(x -> helloRepository.hello(50))
          .flatMap(x -> helloRepository.hello(50))
          .flatMap(x -> helloRepository.hello(50))
          .flatMap(x -> helloRepository.hello(50));
    }

    public Mono<String> multipleHelloZip() {
        return Mono.zip(helloRepository.hello(50),
            helloRepository.hello(50),
            helloRepository.hello(50),
            helloRepository.hello(50),
            helloRepository.hello(50)).map(Tuple2::getT1);
    }
}

File: elixir-ms/lib/domain/usecase/gethello/get_hello_use_case.ex
defmodule ElixirMs.UseCase.GetHelloUseCase do
  @moduledoc false

  @hello_repository Application.compile_env(:elixir_ms, :hello_repository)

  def handle(latency) do
    @hello_repository.hello(latency)
  end

  def handle_multiple() do
    _ = @hello_repository.hello(50)
    _ = @hello_repository.hello(50)
    _ = @hello_repository.hello(50)
    _ = @hello_repository.hello(50)
    @hello_repository.hello(50)
  end
end
